### PR TITLE
Regular (non-react-native) Android Checkout Test

### PR DESCRIPTION
**Pull request by cstavitsky**
- Add 'regular' (non-react-native) checkout test for Android
- Currently the regular Android app is toolstore, not empower plant. The flow is to add several items to the cart and click checkout, triggering an error because there's not enough inventory.
- We run the tests against the build uploaded in the latest Github release from https://github.com/sentry-demos/android
- This PR also contains some refactors of the logic that determines the release version. This logic is now generic and more easily usable by any other language/framework that wants to pull the latest github release from one of the sentry-demos repos.

## Testing

I ran the tests (both `script.sh` and `mobile_native.sh`) locally without issue.

Screen Shot 2021-12-22 at 10 46 59 AM

Screen Shot 2021-12-22 at 10 53 41 AM

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -150,7 +150,41 @@ def android_emu_driver(request, data_center):
     driver.quit()
 
 @pytest.fixture
-def ios_sim_driver(request, data_center):
+def android_emu_driver(request, data_center):
+
+    username_cap = environ['SAUCE_USERNAME']
+    access_key_cap = environ['SAUCE_ACCESS_KEY']
+    release_version = ReleaseVersion.latest_react_native_github_release()
```
Comment (cstavitsky) [inline tests/conftest.py:157]: update to `latest_android_github_release()`

File: tests/conftest.py [hunk 2]
```diff
@@ -150,7 +150,41 @@ def android_emu_driver(request, data_center):
...
+
+    caps = {
+        'username': username_cap,
+        'accessKey': access_key_cap,
+        'deviceName': 'Android GoogleAPI Emulator',
+        'platformVersion': '10.0',
+        'platformName': 'Android',
+        'app': 'https://c66d-157-131-77-226.ngrok.io/app-release.apk',
```
Comment (cstavitsky) [inline tests/conftest.py:165]: update ngrok url to be url to the latest github release

File: tests/mobile_native/android_react_native/test_checkout_react_native_android.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+def test_checkout_react_native_android(android_react_native_emu_driver):
```
Comment (cstavitsky) [inline tests/mobile_native/android_react_native/test_checkout_react_native_android.py:1]: This isn't a 'new' file, I just moved the react native test contents out of `tests/mobile_native/android/test_checkout_android.py` and into a new file.

File: tests/mobile_native/android/test_checkout_android.py [hunk 4]
```diff
@@ -1,25 +1,18 @@
-def test_checkout_android(android_emu_driver):
```
Comment (cstavitsky) [inline tests/mobile_native/android/test_checkout_android.py:1]: This file used to contain the react native android test (for empower plant app). All replacements in the file are for regular android tests (toolstore app).

**Comment by ndmanvar** [comment]
this is amazing Chris (:
👍

**Review by ndmanvar — APPROVED** [review]

**Comment by thinkocapo** [comment]
@ndmanvar @cstavitsky I see under Testing that we ran this locally without issue, but what are we testing for? Link to a transaction? Is this it? I just deployed but I'm not sure what I'm looking for.

https://sentry.io/organizations/testorg-az/discover/paranoid-android:d0416cba0296404695d32c8468eee57f/?field=title&field=event.type&field=project&field=user.display&field=timestamp&name=All+Events&project=1801383&query=&sort=-timestamp&statsPeriod=24h&widths=393&yAxis=count%28%29
```
MainFragment$ItemDeliveryProcessException
 
checkout [android]

Failed to init delivery workflow
```

Is this what we're expecting? Anything more, anything less?
[link](https://sentry.io/organizations/testorg-az/discover/results/?field=title&field=event.type&field=project&field=user.display&field=timestamp&name=All+Events&project=1801383&query=&sort=-timestamp&statsPeriod=24h&widths=393&yAxis=count%28%29)
image

**Pull request merged**